intake/catalog/local.py

```diff
@@ -578,6 +578,23 @@ def _load(self, reload=False):
                 text = text.replace('!template ', '')
             self.parse(text)
 
+    def add(self, **sources):
```

Comment:
I would be inclined to do one `source` at a time here, akin to `set.add`.

Reply:
I had wanted to give an explicit name; but I suppose a source may well have a name, so could be `add(self, source, name=None)` to provide the optional override.